### ISSUE TITLE
Segmentation fault when importing crocoddyl

### ISSUE THREAD

**Issue by PepMS** [loco-3d/crocoddyl#1224]
### Short description: 
when doing 'import crocoddyl' in python, it throws a segmentation fault.

### Long description:
- Ubuntu version: 22
- pinocchio: installed via apt, version 2.7.0
- eigenpy: installed via apt, version 3.3.0
- crocoddyl: branch devel, compiled from source (commit: 4dc442d663b33c2b370d013d5aa5bba8ce356266)

If I debug the segfault with gdb, I get the following text:
```terminal
Program received signal SIGSEGV, Segmentation fault.
0x00005555556745d4 in _Py_Dealloc ()
(gdb) where
#0  0x00005555556745d4 in _Py_Dealloc ()
#1  0x00007ffff0d8beaf in void eigenpy::internal::ExposeStdMethodToStdVector<std::vector<Eigen::Matrix<double, -1, 1, 0, -1, 1>, std::allocator<Eigen::Matrix<double, -1, 1, 0, -1, 1> > >, true, boost::python::def_visitor<eigenpy::EmptyPythonVisitor> >::visit<boost::python::class_<std::vector<Eigen::Matrix<double, -1, 1, 0, -1, 1>, std::allocator<Eigen::Matrix<double, -1, 1, 0, -1, 1> > >, boost::python::detail::not_specified, boost::python::detail::not_specified, boost::python::detail::not_specified> >(boost::python::class_<std::vector<Eigen::Matrix<double, -1, 1, 0, -1, 1>, std::allocator<Eigen::Matrix<double, -1, 1, 0, -1, 1> > >, boost::python::detail::not_specified, boost::python::detail::not_specified, boost::python::detail::not_specified>&) const ()
   from /usr/local/lib/python3.10/dist-packages/crocoddyl/libcrocoddyl_pywrap.cpython-310-x86_64-linux-gnu.so
#2  0x00007ffff0d829c8 in void eigenpy::StdVectorPythonVisitor<std::vector<Eigen::Matrix<double, -1, 1, 0, -1, 1>, std::allocator<Eigen::Matrix<double, -1, 1, 0, -1, 1> > >, true, true, true>::expose<eigenpy::EmptyPythonVisitor>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::python::def_visitor<eigenpy::EmptyPythonVisitor> const&) ()
   from /usr/local/lib/python3.10/dist-packages/crocoddyl/libcrocoddyl_pywrap.cpython-310-x86_64-linux-gnu.so
#3  0x00007ffff0d6c0f9 in crocoddyl::python::init_module_libcrocoddyl_pywrap() ()
   from /usr/local/lib/python3.10/dist-packages/crocoddyl/libcrocoddyl_pywrap.cpython-310-x86_64-linux-gnu.so
#4  0x00007ffff5463a83 in boost::python::detail::exception_handler::operator()(boost::function0<void> const&) const () from /lib/x86_64-linux-gnu/libboost_python310.so.1.74.0
#5  0x00007ffff373dbac in boost::detail::function::function_obj_invoker2<boost::_bi::bind_t<bool, boost::python::detail::translate_exception<eigenpy::Exception, void (*)(eigenpy::Exception const&)>, boost::_bi::list3<boost::arg<1>, boost::arg<2>, boost::_bi::value<void (*)(eigenpy::Exception const&)> > >, bool, boost::python::detail::exception_handler const&, boost::function0<void> const&>::invoke(boost::detail::function::function_buffer&, boost::python::detail::exception_handler const&, boost::function0<void> const&) () from /opt/openrobots/lib/libeigenpy.so
#6  0x00007ffff54626b1 in boost::python::handle_exception_impl(boost::function0<void>) ()
   from /lib/x86_64-linux-gnu/libboost_python310.so.1.74.0
#7  0x00007ffff5463c69 in boost::python::detail::init_module(PyModuleDef&, void (*)()) ()
   from /lib/x86_64-linux-gnu/libboost_python310.so.1.74.0
#8  0x00005555557b3a7a in ?? ()
#9  0x00005555556aec59 in ?? ()
#10 0x00005555556995d7 in _PyEval_EvalFrameDefault ()
#11 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#12 0x000055555569ccfa in _PyEval_EvalFrameDefault ()
#13 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#14 0x000055555569745c in _PyEval_EvalFrameDefault ()
#15 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#16 0x000055555569726d in _PyEval_EvalFrameDefault ()
#17 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#18 0x000055555569726d in _PyEval_EvalFrameDefault ()
#19 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#20 0x000055555569726d in _PyEval_EvalFrameDefault ()
#21 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#22 0x00005555556addc9 in ?? ()
#23 0x000055555578e7cf in _PyObject_CallMethodIdObjArgs ()
#24 0x00005555556c27e3 in PyImport_ImportModuleLevelObject ()
#25 0x0000555555699ea3 in _PyEval_EvalFrameDefault ()
--Type <RET> for more, q to quit, c to continue without paging--
#26 0x00005555556939c6 in ?? ()
#27 0x0000555555789256 in PyEval_EvalCode ()
#28 0x000055555578ee2d in ?? ()
#29 0x00005555556aec59 in ?? ()
#30 0x00005555556995d7 in _PyEval_EvalFrameDefault ()
#31 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#32 0x000055555569ccfa in _PyEval_EvalFrameDefault ()
#33 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#34 0x000055555569745c in _PyEval_EvalFrameDefault ()
#35 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#36 0x000055555569726d in _PyEval_EvalFrameDefault ()
#37 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#38 0x000055555569726d in _PyEval_EvalFrameDefault ()
#39 0x00005555556ae9fc in _PyFunction_Vectorcall ()
#40 0x00005555556addc9 in ?? ()
#41 0x000055555578e7cf in _PyObject_CallMethodIdObjArgs ()
#42 0x00005555556c27e3 in PyImport_ImportModuleLevelObject ()
#43 0x0000555555699ea3 in _PyEval_EvalFrameDefault ()
#44 0x00005555556939c6 in ?? ()
#45 0x0000555555789256 in PyEval_EvalCode ()
#46 0x00005555557b4108 in ?? ()
#47 0x00005555557ad9cb in ?? ()
#48 0x00005555557b3e55 in ?? ()
#49 0x00005555557b3338 in _PyRun_SimpleFileObject ()
#50 0x00005555557b2f83 in _PyRun_AnyFileObject ()
#51 0x00005555557a5a5e in Py_RunMain ()
#52 0x000055555577c02d in Py_BytesMain ()
#53 0x00007ffff7c29d90 in __libc_start_call_main (main=main@entry=0x55555577bff0, argc=argc@entry=2, 
    argv=argv@entry=0x7fffffffdda8) at ../sysdeps/nptl/libc_start_call_main.h:58
#54 0x00007ffff7c29e40 in __libc_start_main_impl (main=0x55555577bff0, argc=2, argv=0x7fffffffdda8, 
    init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffdd98)
    at ../csu/libc-start.c:392
#55 0x000055555577bf25 in _start ()
```
The library worked, but at some point, it stopped working, and this seg fault appeared. I couldn't identify the instant at which this started happening.

I've tried reinstalling Pinocchio and the related packages and recompiling from scratch, but I got the same result. Also, the segfault does not appear when importing Pioncchio or Eigenpy.

I really don't have a clue where to look right now, and I'd appreciate some help.

Thanks!

**Comment by PepMS** [comment]
I've debugged a little bit more. I've confirmed two things:
 1. The error is related to `eigenpy::StdVectorPythonVisitor`. By removing the lines with this visitor, the error disappears.
 2. The error only appears when compiling in Release mode. In Debug, this error does not appear. 

The error might be related to eigenpy, since I replicated this error with another persona library that uses this visitor. Again, in Debug I get no error, while in Release I get a segfault.

I am using the clang compiler
